### PR TITLE
WEB-32461 add hotel_name and rating_class filter with tests and updated docs

### PR DESCRIPTION
**Pull request by kukla1234**
### What
* add hotel_name and rating_class as optional parameters of HotelCollectionRequest
* update documentation to include new parameters
* update HotelCollectionRequest unit test to include new parameters

### How to test
* `composer test-quick` to run unit tests
A full payload that works to return the "Hyatt Regency London - The Churchill" hotel is:
```
$request = new HotelCollectionRequest([
        HotelCollectionRequest::PATH         => 38714,
        HotelCollectionRequest::ITEM         => 5678,
        HotelCollectionRequest::START_DATE   => new DateTime('+1 day'),
        HotelCollectionRequest::END_DATE     => new DateTime('+2 days'),
        HotelCollectionRequest::ROOM_TYPE    => 7,
        HotelCollectionRequest::CURRENCY     => 'EUR',
        HotelCollectionRequest::CATEGORY     => [3,4,5],
        HotelCollectionRequest::LIMIT        => 10,
        HotelCollectionRequest::OFFSET       => 0,
        HotelCollectionRequest::ORDER        => 'price',
        HotelCollectionRequest::RATING_CLASS => [3,4,5],
        HotelCollectionRequest::HOTEL_NAME   => 'Hyatt',
    ]);
$hotels = $tas->getHotelCollection($request);

while (!$hotels->pollingFinished()) {
        $hotels = $tas->getHotelCollection($request);
    }
```

### CODE REVIEW

File: docs/reference/trivago/tas/tas.rst [hunk 1]
```diff
@@ -44,29 +48,33 @@ Request
...
++--------------+-----------+---------+------------+------------------------------------------------------------------------------------------------+
+| CATEGORY     | array     | null    | no         | Hotel categories/rating [0-5].                                                                 |
++--------------+-----------+---------+------------+------------------------------------------------------------------------------------------------+
+| ROOM_TYPE    | int       | null    | no         | Type/size of the room. Possible options: 1 = single-room, 7 = double-room.                     |
++--------------+-----------+---------+------------+------------------------------------------------------------------------------------------------+
+| HOTEL_NAME   | string    | null    | no         | The hotel name. This will search for a hotel with the given name in the area defined by `path`.|
++--------------+-----------+---------+------------+------------------------------------------------------------------------------------------------+
+| RATING_CLASS | array     | null    | no         | Hotel rating class [1-5].                                                                      |
```
Comment (Exeu) [inline docs/reference/trivago/tas/tas.rst]: Can you describe the rating_class a bit more? That an external developer knows what this is doing and what is the difference between CATEGORY.

Comment (kukla1234) [inline docs/reference/trivago/tas/tas.rst]: Yes! Do you know what the difference is? Quite honestly I am not sure, I just took the description from the affiliate docs... though they seem very similar.

Comment (kukla1234) [inline docs/reference/trivago/tas/tas.rst]: okay, got the info, updating now!

**Comment by Kingson-de** [comment]
Please also make SearchParams that are returned in the API response accessible by the SDK.

**Comment by kukla1234** [comment]
@Kingson-de Updated. 

**HotelCollection** and **HotelDeals** both have a `getSearchParams()` method which will return the array of search parameters from the affiliate response. 

Tests and docs updated accordingly. I had to adjust some existing test _assert_ values to use the new/updated fixtures that include search_params

**Comment by Exeu** [comment]
+1

File: src/Response/HotelDeals/HotelDeals.php [hunk 2]
```diff
@@ -29,6 +29,11 @@ class HotelDeals implements \Countable, \Iterator
     private $deals = [];
 
     /**
+     * @var array
+     */
+    private $searchParams = null;
```
Comment (Kingson-de) [inline src/Response/HotelDeals/HotelDeals.php]: you can initialize as an array as described in the docblock

File: src/Response/HotelCollection/HotelCollection.php [hunk 3]
```diff
@@ -30,6 +30,11 @@ class HotelCollection implements \Countable, \Iterator
     private $hotel = [];
 
     /**
+     * @var array
+     */
+    private $searchParams = null;
```
Comment (Kingson-de) [inline src/Response/HotelCollection/HotelCollection.php]: you can initialize as an array as described in the docblock

**Comment by Kingson-de** [comment]
+1

**Pull request merged**